src/gsSolver/gsSimplePreconditioners.h

```diff
@@ -272,7 +306,7 @@ namespace gsGaussSeidel
 ///
 /// \ingroup Solver
 template <typename MatrixType, gsGaussSeidel::ordering ordering = gsGaussSeidel::forward>
-class gsGaussSeidelOp : public gsSteppableOperator<typename MatrixType::Scalar>
+class gsGaussSeidelOp : public gsPreconditionerOp<typename MatrixType::Scalar>
```

Comment:
The free functions gaussSeidelSweep and reverse sweep should be a static function of this class, probably named gsGaussSeidelOp::sweep and gsGaussSeidelOp::revSweep

Reply:
Have removed the free functions (now only in devel)

Only the GaussSeidel free functions are required as the class is based on them. I have moved the remaining 2 free functions to internal namespace, so we can think how to rearrange the code.

Moving the functions to gsGaussSeidelOp is not feasible as that class is templated also in the ordering.